test/distributed/pipeline/sync/skip/test_leak.py

```diff
@@ -15,6 +15,10 @@
 
 @skippable(stash=["skip"])
 class Stash(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.dummy = torch.nn.Parameter(torch.empty(0))
```

Comment:
What is the reason we are adding these dummy params?

Reply:
we need to move these layers to gpu (inorder to prevent merging it with other cpu layers by the new logic). In order to move a module to gpu, it needs to have a parameter.

Reply:
Can we enable this logic only for GPU layers and keep the CPU functionality the same? It doesn't make much sense to merge layers on the CPU.

Reply:
I think it should be consistent. Is there any point in putting consecutive layers that run on cpu on the same host on different partitions? After this optimization, we are telling users that they do not need to worry about partitioning, they just do placement, and we do partitioning based on that. So I think we should not make this extra concern for the user and make them group cpu layers before using our utility.

Reply:
Actually I think merging layers on cpu does help. Consider a case where the whole model is on cpu. Why we should chunk the inputs, and do the scheduling? It will be more efficient if we treat the whole model as one partition in that case.  We should be consistent and have a simple story for the user. With this optimization, we tell the user  not to worry about partitioning, we detect that based on placements.

Reply:
If the whole model is on CPU it actually makes sense to partition the model and pipeline chunks through it. The reason being is that if there is only 1 partition, then only 1 CPU core would be used making it very inefficient. Having multiple partitions means we can have multiple threads executing multiple partitions in parallel making it much more efficient as we utilize all cores.